src/MassTransit/Configuration/JobServiceContainerConfigurationExtensions.cs

```diff
@@ -14,9 +39,18 @@ public static class JobServiceContainerConfigurationExtensions
         /// <returns></returns>
         public static IJobServiceConfigurator ConfigureSagaRepositories(this IJobServiceConfigurator configurator, IServiceProvider provider)
         {
-            configurator.Repository = provider.GetRequiredService<ISagaRepository<JobTypeSaga>>();
-            configurator.JobRepository = provider.GetRequiredService<ISagaRepository<JobSaga>>();
-            configurator.JobAttemptRepository = provider.GetRequiredService<ISagaRepository<JobAttemptSaga>>();
+            static ISagaRepository<T> GetRepository<T>(IServiceProvider provider)
+                where T : class, ISaga
+            {
+                var factory = new DependencyInjectionSagaRepositoryContextFactory<T>(provider, LegacySetScopedConsumeContext.Instance);
```

Comment:
Same thing as the other files, I really need to understand _why_ this is being replicated everywhere. As in, I don't like it.